src/allmydata/web/common_py3.py

```diff
@@ -23,24 +15,18 @@ def __init__(self, text, code=http.BAD_REQUEST):
         self.code = code
 
 
-def get_arg(ctx_or_req, argname, default=None, multiple=False):
+def get_arg(req, argname, default=None, multiple=False):
     """Extract an argument from either the query args (req.args) or the form
     body fields (req.fields). If multiple=False, this returns a single value
     (or the default, which defaults to None), and the query args take
     precedence. If multiple=True, this returns a tuple of arguments (possibly
     empty), starting with all those in the query args.
     """
     results = []
-    if PY2:
-        req = INevowRequest(ctx_or_req)
-        if argname in req.args:
-            results.extend(req.args[argname])
-        if req.fields and argname in req.fields:
-            results.append(req.fields[argname].value)
-    else:
-        req = IRequest(ctx_or_req)
-        if argname in req.args:
-            results.extend(req.args[argname])
+    if argname in req.args:
+        results.extend(req.args[argname])
+    if req.fields and argname in req.fields:
```

Comment:
This `fields` here is application specific (as in, specific to `TahoeLAFSRequest`), right?

I think `get_args` would be less general since `twisted.web.server.Request` is not supposed to have a `fields` attribute, although I am not sure if that would actually be a problem.

Reply:
> This fields here is application specific (as in, specific to TahoeLAFSRequest), right?

> I think get_args would be less general since twisted.web.server.Request is not supposed to have a fields attribute, although I am not sure if that would actually be a problem.

￼I'm not sure about this.  Less general than what?

Reply:
I meant that `get_root` won't be able to handle a `Request` or `DummyRequest`, because they don't have a `field` attribute.  For example, if someone is adding a test that uses one of those things and if that test hits `get_root()`, they will need to use `TahoeLAFSRequest`.

I am not sure if that would be a less preferable outcome though.

Reply:
Ah, I see.  Yea, that seems correct.  Also seems fine, I think.  Perhaps everyone can use the `render` helper which provides the right kind of request.  At least, we can try that for a while.